src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -677,7 +678,7 @@ def create_python_udf_or_sp(
 
     create_query = f"""
 CREATE{" OR REPLACE " if replace else ""}
-{"TEMPORARY" if is_temporary else ""} {object_type.value} {object_name}({sql_func_args})
+{"TEMPORARY" if is_temporary else ""} {"SECURE" if secure else ""} {object_type.value} {object_name}({sql_func_args})
```

Comment:
A space may be needed between "TEMPORARY" and "SECURE".

Reply:
There is a space already since this is inside `"""` block :p